src/azure-cli-core/azure/cli/core/commands/arm.py

```
@@ -312,6 +324,106 @@ def __call__(self, parser, namespace, values, option_string=None):
+                        return
+                if custom_condition and bool(verify_property(instance, custom_condition)):
+                    return
+            except ClientException as ex:
+                if getattr(ex, 'status_code', None) == 404:
+                    if wait_for_deleted:
+                        return
+                    if not any([wait_for_created, wait_for_exists, custom_condition]):
+                        raise
+                else:
+                    raise
+
+            time.sleep(interval)
+
+        return CLIError('Timeout')
+
+    cmd = CliCommand(name, handler, table_transformer=table_transformer,
+                     arguments_loader=arguments_loader)
+    group_name = 'Wait Condition'
+    cmd.add_argument('timeout', '--timeout', default=3600, arg_group=group_name, type=int,
+                     help='maximum wait in seconds')
+    cmd.add_argument('interval', '--interval', default=30, arg_group=group_name, type=int,
+                     help='polling interval in seconds')
+    cmd.add_argument('deleted', '--deleted', action='store_true', arg_group=group_name,
+                     help='wait till deleted')
+    cmd.add_argument('created', '--created', action='store_true', arg_group=group_name,
+                     help="wait till created with 'provisioningState' at 'Succeeded'")
+    cmd.add_argument('updated', '--updated', action='store_true', arg_group=group_name,
+                     help="wait till updated with provisioningState at 'Succeeded'")
+    cmd.add_argument('exists', '--exists', action='store_true', arg_group=group_name,
+                     help="wait till the resource exists")
+    cmd.add_argument('custom', '--custom', arg_group=group_name,
+                     help=("Wait until the condition satisfies a custom JMESPath query. E.g. "
+                           "provisioningState!='InProgress', "
+                           "instanceView.statuses[?code=='PowerState/running']"))
```

I think it does that at line 423 essentially. It compiles the JMESPath, which should raise an error if it is invalid.